File: src/etch/bindings/java/msg/Validator_boolean.java
package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for boolean.
 */
public class Validator_boolean extends TypeValidator
{
	/**
	 * @param nDims the number of dimensions. 0 for a scalar.
	 * @return an instance of the validator.
	 */
	public static Validator_boolean get( int nDims )
    {
		if (nDims < 0)
			return null;
		
		if (nDims >= validators.length)
			return new Validator_boolean( nDims );
		
		if (validators[nDims] == null)
			validators[nDims] = new Validator_boolean( nDims );
		
		return validators[nDims];
    }
	
	private static Validator_boolean[] validators = new Validator_boolean[4];
	
	/**
	 * Constructs the Validator.
	 *
	 * @param nDims the number of dimensions. 0 for a scalar.
	 */
	private Validator_boolean( int nDims )
	{
		super( Boolean.class, boolean.class, nDims, "boolean["+nDims+"]" );
	}

	@Override
    public Validator elementValidator()
    {
	    return get( nDims-1 );
    }

	@Override
	public Byte checkValue( Object value )
	{
		if (!validate( value))
			return null;
		
		if (nDims > 0)
			return TypeCode.ARRAY;
		
		Boolean v = (Boolean) value;
		
		return v ? TypeCode.BOOLEAN_TRUE : TypeCode.BOOLEAN_FALSE;
	}
}

File: src/etch/bindings/csharp/msg/Validator_boolean.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for Boolean
    /// </summary>
    public class Validator_boolean : TypeValidator
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="nDims">number of dimensions. 0 for a scalar.</param>
        /// <returns>an instance of the validator</returns>
        public static Validator_boolean Get( int nDims )
        {
            CheckDims( nDims );

            if ( nDims >= validators.Length )
                return new Validator_boolean( nDims );

            Validator_boolean v = validators[ nDims ];

            if ( v == null )
                v = validators[ nDims ] = new Validator_boolean( nDims );

            return v;
        }

        private static Validator_boolean[] validators = new Validator_boolean[ MAX_CACHED ];

        /// <summary>
        /// Constructs the validator.
        /// </summary>
        /// <param name="nDims"></param>
        private Validator_boolean( int nDims )
            : base( typeof( bool ), typeof( bool ), nDims, "boolean["+nDims+"]" )
        { 
        }


        public override Validator ElementValidator()
        {
            return Get( nDims - 1 );
        }

        public override sbyte? CheckValue( object value )
        {
            if ( !Validate( value ) )
                return null;

            if ( nDims > 0 )
                return Etch.Transport.Fmt.TypeCode.ARRAY;

            Boolean v = ( Boolean ) value;
            return v ? Etch.Transport.Fmt.TypeCode.BOOLEAN_TRUE : Etch.Transport.Fmt.TypeCode.BOOLEAN_FALSE;
        }
    }
}
